app/src/main/java/com/example/awesomelanguagelearning/forgot_password/ui/ForgotPasswordViewModel.kt

```diff
@@ -0,0 +1,37 @@
+package com.example.awesomelanguagelearning.forgot_password.ui
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.awesomelanguagelearning.forgot_password.ui.models.ForgotPasswordState
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+
+class ForgotPasswordViewModel: ViewModel() {
+
+    private val _emailStateFlow = MutableStateFlow(ForgotPasswordState())
+    val emailStateFlow = _emailStateFlow.asStateFlow()
+
+    private val _effect = Channel<Unit>()
+    val effect = _effect.receiveAsFlow()
+
+    fun updateEmail(email: String) {
+        viewModelScope.launch {
```

Comment:
@igel911 I think launching coroutine is kind of redundant here and you can call `_emailStateFlow.update {}` without it, because `update` is not a suspendable method

Reply:
@slothdeboss done